apps/emqx_connector/src/emqx_connector_pgsql.erl

```diff
@@ -344,6 +336,30 @@ do_check_prepares(State = #{pool_name := PoolName, prepare_sql := {error, Prepar
             {error, Error}
     end.
 
+-spec validate_table_existence([pid()], binary()) -> ok | {error, undefined_table}.
+validate_table_existence([WorkerPid | Rest], SQL) ->
+    try ecpool_worker:client(WorkerPid) of
+        {ok, Conn} ->
+            case epgsql:parse2(Conn, "", SQL, []) of
+                {error, {_, _, _, undefined_table, _, _}} ->
+                    {error, undefined_table};
+                Res when is_tuple(Res) andalso ok == element(1, Res) ->
+                    ok;
+                Res ->
+                    ?tp(postgres_connector_bad_parse2, #{result => Res}),
+                    validate_table_existence(Rest, SQL)
+            end;
+        _ ->
+            validate_table_existence(Rest, SQL)
+    catch
+        exit:{noproc, _} ->
+            validate_table_existence(Rest, SQL)
+    end;
+validate_table_existence([], _SQL) ->
+    %% All workers either replied an unexpected error; we will retry
```

Comment:
```suggestion
    %% All workers replied an unexpected error; we will retry
```

Reply:
if all workers replied error, shouldn't we already move the bridge to `disconnected`?

Reply:
We are just checking for table existence.  It'll transition to disconnected only if we are certain that the table does not exist, as was the behavior before.  They may have just timed out, so we defer the check.